### DISCUSSION TITLE
Laravel dd (die dump) is not (really) working

### DISCUSSION THREAD

**Discussion by TheArKaID** [php/frankenphp#1021] [Q&A] [answered]
Usually, or in my local running with nginx php-fpm, when a request is respoding with `dd()`, this will returned.

![image](https://github.com/user-attachments/assets/d04530f9-d6f6-45b6-a859-127933e46f73)

But now, when I run in FrankenPHP (without Nginx), an error in console is appear but the http request is done:

![image](https://github.com/user-attachments/assets/94bad043-58ce-4120-9923-cb4d9f181fd6)

And the process indicator (opacity 50%) is not changed. Is that how it should be?

**Comment by AlliBalliBaba** [comment]
I noticed the same, the problem with `dd()` in Laravel Octane is that it will return a 200 status code, while without worker mode it will return a 500 status code. The 500 status code is needed for javascript to notice that the response is not a regular response.

You can open an issue at their [github](https://github.com/laravel/octane/issues), maybe it's fixable, but they have closed [similar issues](https://github.com/laravel/octane/issues/838) with `dump()` before.

**Reply by dunglas** [reply]
Do you know if this problem occurs only with FrankenPHP or it affects all Octane engines?

**Reply by TheArKaID** [reply]
Thanks for mentioning your reports, it makes me retrying it. In my case, either `dump()` or `dd()` would work perfectly in View:

![image](https://github.com/user-attachments/assets/7d6217c8-1a12-48e3-b706-5785bb9e748d)

But if the `dump()` followed by `dd()`, `dd()` would replace the whole view including the `dump()`. Usually I will get the html ended with the result of `dd()`, like this:

![image](https://github.com/user-attachments/assets/c20e5cf5-2a51-4de8-b296-d3d95ca9daa3)

Before that, I forgot to mention that I'm using Livewire, the `dd()` from controller works perfectly:
![image](https://github.com/user-attachments/assets/d4bcac47-3f95-4cf0-b687-803767eb151b)

The only unexpected behavior is when Livewire http request responded with `dd()`. And now I'm confusing whether to try reporting it to Octane or Livewire.

**Reply by AlliBalliBaba** [reply]
It affects all Octane engines, not just Frankenphp

**Reply by AlliBalliBaba** [reply]
And it's definitely an Octane issue, the usual `dd()` from symfony uses an `exit` statement, which Swoole can't handle, so they have an alternate approach.
Thinking about it, does Frankenphp handle `exit` regularly in worker mode @dunglas ?

**Reply by withinboredom** [reply]
IIRC @AlliBalliBaba, calling `exit` while in a worker request just ends the request (and flushes output).

**Reply by AlliBalliBaba** [reply]
Hmm in that case maybe I'll attempt a PR to fall back to the original behavior on frankenphp

**Reply by AlliBalliBaba** [reply, marked as answer]
This [should be fixed](https://github.com/laravel/octane/pull/945) once octane releases a new version (`2.5.5` or `2.6.0`)

**Reply by TheArKaID** [reply]
Thanks @AlliBalliBaba
